src/ast/mod.rs

```diff
@@ -476,14 +476,18 @@ pub enum Expr {
     },
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
```

Comment:
Seeing this makes me realize, it has long stopped making sense for ArgEncoding to impl Hash.  You can remove the Hash impls from this, ArgEncoding, and StringSize.

Reply:
Same with `Eq`

Reply:
`Eq` appears to be necessary for the `Ord` trait. The compiler throws a fit if it gets removed.

Reply:
StringArgSize shouldn't be Ord either, that's a mistake

Reply:
And I'm not sure what asserts you're referring to, I removed it from my local copy just fine.  assert_eq! requires only PartialEq

Reply:
It was some sort of error about "required by bound in AssertParamIsEq" or something like that. However, once I removed `Ord` from StringArgSize it stopped complaining about both types, so I guess it's fine now?